src/components/cc-range-selector-option/cc-range-selector-option.js

```diff
@@ -0,0 +1,193 @@
+import { LitElement, css, html } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+
+/**
+ * A tile component that can be used to display a range selection state.
+ *
+ * This component is specifically designed for cc-range-selector and is not meant to be used standalone.
+ * It is a presentational component that displays visual states based on properties set by its parent.
+ * All interactive behaviors (click handling, keyboard navigation, etc.) are managed by the parent cc-range-selector component.
+ *
+ * @cssdisplay inline-flex
+ *
+ * @slot body - Content displayed as the main part of the tile. This slot should contain the option's label or visual content and should not be empty.
+ */
+export class CcRangeSelectorOption extends LitElement {
+  static get properties() {
+    return {
+      disabled: { type: Boolean, reflect: true },
+      dragging: { type: Boolean, reflect: true },
+      error: { type: Boolean, reflect: true },
+      pointer: { type: Boolean, reflect: true },
+      readonly: { type: Boolean, reflect: true },
+      selected: { type: Boolean, reflect: true },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {boolean} Whether the component should be disabled (default: 'false') */
+    this.disabled = false;
+
+    /** @type {boolean} Whether the option is within a drag selection range (default: 'false') */
+    this.dragging = false;
+
+    /** @type {boolean} Whether the component should be in error mode when not disabled nor readonly (default: 'false') */
+    this.error = false;
+
+    /** @type {boolean} Whether to show a pointer cursor for interactive states (default: 'false') */
+    this.pointer = false;
+
+    /** @type {boolean} Whether the component should be readonly when not disabled (default: 'false') */
+    this.readonly = false;
+
+    /** @type {boolean} Whether the option is currently selected when not dragging (default: 'false') */
+    this.selected = false;
+  }
+
+  /**
+   * Renders the option with appropriate visual states based on its properties.
+   * Calculates CSS classes for disabled, readonly, error, dragging, selected, and pointer states.
+   * @return {import('lit').TemplateResult}
+   */
+  render() {
+    // Calculate CSS classes based on component state
+    // State priority: disabled > readonly > error > dragging > selected
+    // Note: dragging state takes visual priority over selected state during drag operations
+    const classes = {
+      disabled: this.disabled,
+      readonly: !this.disabled && this.readonly,
+      error: !this.disabled && !this.readonly && this.error,
+      dragging: this.dragging,
+      selected: !this.dragging && this.selected, // Selected styling is hidden during dragging
+    };
+
+    return html`
+      <div class="wrapper ${classMap(classes)}">
+        <slot name="body"></slot>
```

Comment:
question: do we really need a named slot when there is only one slot? Why not using the default slot instead?

Reply:
You're right, it's a relic from old iterations: I'll remove it, it will make the code simpler. :+1: